### PR TITLE
Do not interpret quotes as string literals inside string constructor

### LINKED ISSUE #1317: Problem with string constructor interpolations surrounded by quotes
### What is the problem

Support for XQuery 3.1 string constructors (https://www.w3.org/TR/xquery-31/#id-string-constructors) was added in https://github.com/eXist-db/exist/pull/1297. Running the current develop branch, I get an error when surrounding an interpolation with quotes.

### What did you expect

No error with the following code:

### How to reproduce or add a test

```xquery
xquery version "3.1";

let $a := "my"
return
    ``[Hello "`{$a}`" world!]``
```

or as an xqsuite test:

```xquery
xquery version "3.1";

module namespace sc="http://exist-db.org/xquery/test/string-constructor";

declare namespace test="http://exist-db.org/xquery/xqsuite";

declare
    %test:assertEquals('Hello "my" world!')
function sc:simple-interpolation-surrounded-by-quotes() {
    let $a := "my"
    return
        ``[Hello "`{$a}`" world!]``
};
```

This yields the error:

```
Cannot compile xquery: exerr:ERROR expecting '"', found '<EOF>' [at line 12, column 39] 
```

BaseX 8.6 and Saxon 9.7.0.4 return the expected results: `Hello "my" world!`

### Context information

- eXist-db version: eXist-db 3.0 develop 
- Java version: Java 1.8.0_121-b13
- Operating system: macOS
- 64 bit
- Any custom changes in e.g. conf.xml: no

### PR DESCRIPTION
**Pull request by wolfgangmm**
Make sure special chars do not need to be escaped and character entities are not expanded.

Closes #1317

### CODE REVIEW

File: src/org/exist/xquery/parser/XQuery.g [hunk 1]
```diff
@@ -2484,10 +2484,10 @@ options {
 	APOS_ATTRIBUTE_CONTENT
 	{ $setType(APOS_ATTRIBUTE_CONTENT); }
 	|
-	{ !(parseStringLiterals || inElementContent) }?
+	{ !(parseStringLiterals || inElementContent || inStringConstructor) }?
```
Comment (shabanovd) [inline src/org/exist/xquery/parser/XQuery.g]: where is `inStringConstructor` set?

**Comment by shabanovd** [comment]
What do `archive` mean here?

File: src/org/exist/xquery/parser/XQuery.g (hunk 1 above)
Comment (wolfgangmm) [inline src/org/exist/xquery/parser/XQuery.g]: [line 2096](https://github.com/wolfgangmm/exist/blob/feafc07015b1ac2cb93fe851ae93b055a260bf98/src/org/exist/xquery/parser/XQuery.g#L2096)

**Comment by joewiz** [comment]
Looks great! Thank you @wolfgangmm!

**Pull request merged**